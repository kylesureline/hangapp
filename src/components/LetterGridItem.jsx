import React from 'react';
import { connect } from 'react-redux';

export const LetterGridItem = (props) => (
  <button onClick={props.handleGuess} disabled={props.guessedLetters.indexOf(props.letter) !== -1}>{props.letter}</button>
);

const mapStateToProps = (state) => ({
  guessedLetters: state.player.guessedLetters
});

export default connect(mapStateToProps)(LetterGridItem);
