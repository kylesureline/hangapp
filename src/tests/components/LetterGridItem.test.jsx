import React from 'react';
import { shallow } from 'enzyme';
import { LetterGridItem } from '../../components/LetterGridItem.jsx';

test('should render LetterGridItem not disabled correctly', () => {
  const wrapper = shallow(<LetterGridItem letter="a" guessedLetters={['z']} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render LetterGridItem disabled correctly', () => {
  const wrapper = shallow(<LetterGridItem letter="z" guessedLetters={['z']} />);
  expect(wrapper).toMatchSnapshot();
});

test('should handleGuess correctly', () => {
  const handleGuess = jest.fn();
  const wrapper = shallow(<LetterGridItem handleGuess={handleGuess} letter="a" guessedLetters={['z']} />);
  wrapper.find('button').simulate('click');
  expect(handleGuess).toHaveBeenCalled();
});
