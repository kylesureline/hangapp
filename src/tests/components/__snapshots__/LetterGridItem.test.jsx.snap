// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LetterGridItem disabled correctly 1`] = `
<button
  className="button button--letter"
  disabled={true}
>
  z
</button>
`;

exports[`should render LetterGridItem not disabled correctly 1`] = `
<button
  className="button button--letter"
  disabled={false}
>
  a
</button>
`;
